reco_utils/common/tf_utils.py

```diff
@@ -0,0 +1,408 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import abc
+import itertools
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+
+from reco_utils.common.constants import (
+    DEFAULT_USER_COL,
+    DEFAULT_ITEM_COL
+)
+
+
+MODEL_DIR = 'model_checkpoints'
+
+
+def pandas_input_fn(
+    df,
+    y_col=None,
+    batch_size=128,
+    num_epochs=1,
+    shuffle=False,
+    num_threads=1
```

Comment:
I'm usually a fan of using all the cores of a machine. I use this function https://github.com/miguelgfierro/codebase/blob/master/python/system/system_info.py#L76. Is there a reason why is it better to use only 1 thread?

Reply:
The thread here is for data reading and enqueueing. Here is the document from TF:

> number of threads used for reading and enqueueing. In order to have predicted and repeatable order of reading and enqueueing, such as in prediction and evaluation mode, num_threads should be 1.

For this reason, the default value is  1 as well as `shuffle=False`. A user can set, e.g., `num_threads=num_cores-1` for training. Will change that on wide-deep notebook.